tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/process/traversal/step/sideEffect/TinkerGraphStep.java

```diff
@@ -132,8 +135,20 @@ private <E extends Element> Iterator<E> iteratorList(final Iterator<E> iterator)
         try {
             while (iterator.hasNext()) {
                 final E e = iterator.next();
-                if (HasContainer.testAll(e, this.hasContainers))
-                    list.add(e);
+                try {
+                    if (HasContainer.testAll(e, this.hasContainers))
+                        list.add(e);
+                } catch (GremlinTypeErrorException ex) {
+                    if (getTraversal().isRoot() || !(getTraversal().getParent().asStep() instanceof FilterStep)) {
```

Comment:
probably better to do not catch/throw exception for non-root and FilterStep.
code might look like
```
if (getTraversal().isRoot() || !(getTraversal().getParent().asStep() instanceof FilterStep)) {
  try { doWork(); } catch (GremlinTypeErrorException ex) {}
} else { 
  doWork();
}
```

Reply:
Thanks for the suggestion, I have refactored the code accordingly.